hoomd/hpmc/update.py

```diff
@@ -512,6 +513,207 @@ def N(self):  # noqa: N802 - allow N as a function name
+
+    Examples::
+
+        mc = hoomd.hpmc.integrate.ConvexPolyhedron(23456)
+        mc.shape["A"] = dict(vertices=numpy.asarray([(1, 1, 1), (-1, -1, 1),
+                                                    (1, -1, -1), (-1, 1, -1)]) / 2)
+        vertex_move = hoomd.hpmc.shape_move.Vertex(stepsize={'A': 0.01},
+                                                   param_ratio=0.2,
+                                                   volume=1.0)
+        updater = hoomd.hpmc.update.Shape(shape_move=vertex_move,
+                                          trigger=hoomd.trigger.Periodic(1),
+                                          nselect=1,
+                                          nsweeps=1,
+                                          multi_phase=False,
+                                          num_phase=1)
+
+    Attributes:
+        trigger (Trigger): Call the updater on triggered time steps.
+
+        step_size (`TypeParameter` [``particle type``, `float`]): Maximum size
+            of shape trial moves.
+
+        shape_move (ShapeMove): Type of shape move to apply when updating shape
+            definitions
+
+        pretend (bool): When True the updater will not actually update the shape
+            definitions, instead moves will be proposed and the acceptance
+            statistics will be updated correctly.
+
+        nselect (int): Number of types to change every time the updater is
+            called.
+
+        nsweeps (int): Number of times to update shape definitions during each
+            triggered timesteps.
+
+        multi_phase (bool): When True MPI is enforced and shapes are updated
+            together for two boxes.
+
+        num_phase (int): How many boxes are simulated at the same time,
+            supported values are 1, 2 and 3.
+    """
+
+    def __init__(self,
+                 trigger,
+                 step_size=0.2,
+                 pretend=False,
+                 nselect=1,
+                 nsweeps=1,
+                 multi_phase=False,
+                 num_phase=1):
+        super().__init__(trigger)
+        param_dict = ParameterDict(shape_move=hoomd.hpmc.shape_move.ShapeMove,
+                                   pretend=bool(pretend),
+                                   nselect=int(nselect),
+                                   nsweeps=int(nsweeps),
+                                   multi_phase=bool(multi_phase),
+                                   num_phase=int(num_phase))
+        self._param_dict.update(param_dict)
+
+        typeparam_step_size = TypeParameter('step_size',
+                                            type_kind='particle_types',
+                                            param_dict=TypeParameterDict(
+                                                float(step_size), len_keys=1))
+
+        self._extend_typeparam([typeparam_step_size])
+
+    def _add(self, simulation):
+        super()._add(simulation)
+        self.shape_move._add(simulation)
```

Comment:
This logic needs to handle additional cases such as attempting to the same move object in two simulations. See the `Pair.nlist` logic for a reference.